File: primo-anno/Programmazione/python/sortedarray.py
#scrivere in python una funzione che dato un array restituisce TRUE se è strettamente 
#crescente, FALSE altrimenti

A = [1, 2, 3, 4, 5, 6, 7, 10]

def sortedarray(A): 
    for i in range(len(A)-1): 
        if A[i] > A[i+1]: 
            return False
    
    return True 

print(sortedarray(A))

#complessità temporale O(n), complessità spaziale O(1)

File: primo-anno/Programmazione/C/sortedarray.c
//scrivere una funzione in C che dato un vettore, determina se esso è ordinato in modo crescente
#include <stdio.h>
#include <stdbool.h>

bool sortedarray(int *A, int sizeA){
    
    for(int i = 1; i < sizeA; i++){

        if(A[i-1]>A[i]){
            return false; 
        }
    }

    return true; 
}

int main(){
    int A[] = {1, 2, 4, 3 ,1, 34}; 
    int sizeA = sizeof(A)/sizeof(A[0]); 
    printf("%d", sortedarray(A, sizeA)); 

}